devices/pine64-pinephone-braveheart/kernel/default.nix

```
@@ -6,21 +6,22 @@
 }:
 
 (mobile-nixos.kernel-builder {
-  version = "5.5.0";
+  version = "5.6.0";
   configfile = ./config.aarch64;
   src = fetchFromGitLab {
     owner = "pine64-org";
     repo = "linux";
-    rev = "94cf851f0f4443c771a926102dee497def319b49";
-    sha256 = "1a4ch2j8hla3xd7rv38ra6bnv14lsnj0srhlh1c8vxxvwywzg815";
+    rev = "2a6074e53b019db8a4a63cb255bc29422db3fe19";
+    sha256 = "1rjs2wvbbq69mzr6i3hjdbwsr61gflzvgx13z5hf89gcp29idmcp";
   };
   patches = [
-    ./0001-dts-pinephone-Setup-default-on-and-panic-LEDs.patch
+     ./0001-dts-pinephone-Setup-default-on-and-panic-LEDs.patch
+     ./disable-power-save.patch
   ];
 }).overrideAttrs({ postInstall ? "", ... }: {
   installTargets = [ "install" "dtbs" ];
   postInstall = postInstall + ''
     mkdir -p "$out/dtbs/allwinner"
-    cp -v "$buildRoot/arch/arm64/boot/dts/allwinner/sun50i-a64-pinephone.dtb" "$out/dtbs/allwinner/"
+    cp -v "$buildRoot/arch/arm64/boot/dts/allwinner/sun50i-a64-pinephone-1.1.dtb" "$out/dtbs/allwinner/sun50i-a64-pinephone.dtb"
```

This is a hard problem to solve.

U-Boot will, at some point, be updated to one that detects between 1.1 and 1.2, and at that moment it will need the appropriately "1.1" filename.

I think the solution, for the time being, will be to copy it both to the new name, and to the old name at that location. This way the kernel build will stay compatible with previous U-Boot builds, while allowing support with the new one.


```suggestion
    # Filename expected by older U-Boot revisions for the Pinephone.
    cp -v "$buildRoot/arch/arm64/boot/dts/allwinner/sun50i-a64-pinephone-1.1.dtb" "$out/dtbs/allwinner/sun50i-a64-pinephone.dtb"
    # Hardware-versioned filenames.
    cp -v "$buildRoot/arch/arm64/boot/dts/allwinner/sun50i-a64-pinephone-1.1.dtb" "$out/dtbs/allwinner/"
```

---

Btw, there is also a `1.0.dtb`. Should we add it as well?

---

Yes, please. Though I'm not sure it'll be useful for anyone, we should strive to be a good example in maximizing compatibility.

---

done!